woocommerce/woocommerce
# Deprecate All Report classes & packages

_This is a part of https://github.com/woocommerce/woocommerce/pull/51665_

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





- [Deprecate All Report classes & packages in PHPDoc](https://github.com/woocommerce/woocommerce/commit/ab17231ed5183d1b53eab1de27effdf4c75263aa)
- [Throww deprecation warnings for Report classes' constructors](https://github.com/woocommerce/woocommerce/commit/89efd7ea846e33e4d005f8fefca82bf422dfca9e)

The following classes are indirectly deprecated:
- `plugins/woocommerce/includes/rest-api/Controllers/Version1/class-wc-rest-report-sales-v1-controller.php`
- `plugins/woocommerce/includes/rest-api/Controllers/Version1/class-wc-rest-report-top-sellers-v1-controller.php`
- `plugins/woocommerce/includes/rest-api/Controllers/Version1/class-wc-rest-reports-v1-controller.php`

but not directly deprecate them, as we plan to decouple them from Reports

I didn't add deprecation warnings to `WC_Report_Sales_By_Date` and `WC_Admin_Reports` not to throw them in REST controller as we still going to support it, we may add those warnings once we decouple it.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Review the code checking that all methods and classes are deprecated.
2. Generate [the `code-reference` ](https://github.com/woocommerce/code-reference) for this branch 
	1. `pnpm i && pnpm run build && pnpm run build:zip` here
	2. Unpack the zip to code-reference folder
	3. `./deploy.sh -v --build-only -s 9.3.0 --no-download` there
3. Check that the classes and methods are marked as deprecated
   ![image](https://github.com/user-attachments/assets/87961ccf-8132-4da3-9f08-0e07417c7d96)
5. Visit `/wp-admin/admin.php?page=wc-reports`
   All reports except "Sales by date" should show deprecation warning.
   ![image](https://github.com/user-attachments/assets/b62085d7-f2e0-4f18-90fe-1141cbbe209e)
   ![image](https://github.com/user-attachments/assets/1a1251d9-8831-4b3c-8944-6e84976690de)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
